server/common/config/client_config.py

```diff
@@ -29,21 +41,28 @@ def get_client_config(app_config: AppConfig, data_adaptor: Dataset, current_app)
         "about-dataset": about,
     }
 
+    auth_disabled = app_config.server__app__disable_auth
+
     # parameters
     parameters = {
         "layout": dataset_config.default_dataset__embeddings__names,
         "max-category-items": dataset_config.default_dataset__presentation__max_categories,
         "diffexp_lfc_cutoff": dataset_config.default_dataset__diffexp__lfc_cutoff,
         "disable-diffexp": not dataset_config.default_dataset__diffexp__enable,
-        "annotations_genesets": True,  # feature flag
-        "annotations_genesets_readonly": True,
+        "annotations": not auth_disabled,  # Disable annotations when auth is disabled
+        "annotations_genesets": not auth_disabled,  # Disable genesets when auth is disabled
+        "annotations_genesets_readonly": auth_disabled,  # Make readonly when auth is disabled
         "annotations_genesets_summary_methods": ["mean"],
         "custom_colors": dataset_config.default_dataset__presentation__custom_colors,
         "diffexp-may-be-slow": False,
         "about_legal_tos": dataset_config.default_dataset__app__about_legal_tos,
         "about_legal_privacy": dataset_config.default_dataset__app__about_legal_privacy,
     }
 
+    # Provide autosave-related client parameters. These defaults may be overridden by
+    # the adaptor's update_parameters implementation when richer configuration exists.
+    parameters.setdefault("annotations-user-data-idhash", _compute_user_idhash(data_adaptor))
```

Comment:
annotations-user-data-idhash derived from dataset location (blake2b). Dataset-scoped, no PII. Confirm sufficiency for client autosave identity and lack of collisions across dataroots.